go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -579,21 +579,23 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 	// Note: for ovnkube node mode dpu-host no metrics is required as ovs/ovn is not running on the node.
 	if config.OvnKubeNode.Mode != types.NodeModeDPUHost && config.Metrics.OVNMetricsBindAddress != "" {
 		metricsScrapeInterval := 30
-		defer cancel()
 
 		if ovsClient == nil {
 			ovsClient, err = libovsdb.NewOVSClient(ctx.Done())
 			if err != nil {
-				return fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+				ovsCLIErr = fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+				cancel()
```

Comment:
oh we were returning here from entire `runOvnKube`... w/o even handling the errors from other goroutines!